src/netcdf_writer.jl

```diff
@@ -142,6 +142,18 @@ function NetCDFWriter(
         )
     end
 
+    if is_horizontal_space
```

Comment:
@Sbozzolo Note we have two conditional statement `if_horizontal_space` in a row, we could possible merge them together? 

We could also move `comms_ctx = ClimaComms.context(space)` above the if statement?